### ISSUE TITLE
Zipline setup.py always unconditionally installs numpy 1.9

### ISSUE THREAD

**Issue by ssanderson** [quantopian/zipline#954]
```
(zipline)[~/quantopian/zipline]@(master:0dac2e056f)$ pip install numpy==1.10
...
Successfully installed numpy-1.10.0
(zipline)[~/quantopian/zipline]@(master:0dac2e056f)$ pip install -e .
...
Installing collected packages: Logbook, pytz, numpy, scipy, six, python-dateutil, pandas, patsy, statsmodels, requests, cyordereddict, bottleneck, contextlib2, decorator, networkx, numexpr, bcolz, click, toolz, sqlalchemy, zipline
  Found existing installation: numpy 1.10.0
    Can't uninstall 'numpy'. No files were found to uninstall.
  Running setup.py develop for zipline
Successfully installed Logbook-0.12.5 bcolz-0.12.1 bottleneck-1.0.0 click-6.2 contextlib2-0.5.1 cyordereddict-0.2.2 decorator-4.0.6 networkx-1.10 numexpr-2.4.6 numpy-1.9.2 pandas-0.17.1 patsy-0.4.1 python-dateutil-2.4.2 pytz-2015.7 requests-2.9.1 scipy-0.16.1 six-1.10.0 sqlalchemy-1.0.11 statsmodels-0.6.1 toolz-0.7.4 zipline
(zipline)[~/quantopian/zipline]@(master:0dac2e056f)$ python
Python 2.7.10 (default, Jun 30 2015, 15:30:23)
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> numpy.__version__
'1.9.3'
>>>

**Comment by ssanderson** [comment]
This is likely the cause of https://github.com/quantopian/pyfolio/pull/272

**Comment by richafrank** [comment]
I thought we intentionally didn't support >1.9 yet because of https://github.com/quantopian/zipline/issues/894#issuecomment-162696875

**Comment by ssanderson** [comment]
If the user already has numpy 1.10 installed, I don't think we should be downgrading them.  If we **do** downgrade them, we certainly shouldn't do so silently.  This will break any other packages they had previously compiled against numpy 1.10, often in nasty ways.

**Comment by twiecki** [comment]
How difficult would it be to make zipline numpy 1.10 compatible? That problem won't go away but should unlock a few things if fixed (like usable conda packages).

**Comment by ssanderson** [comment]
@richafrank it looks like this is unintended behavior at any rate.  We should be passing `strict_bounds=False` in the pre-setup code, but it looks like there's a bug in that logic.

**Comment by ssanderson** [comment]
@twiecki the issue affecting us in numpy has been fixed as of https://github.com/numpy/numpy/pull/6465, which is slated to be released in the upcoming 1.11.  It might be easier at this point to just skip 1.10 entirely and support 1.11.

**Comment by twiecki** [comment]
I see, that makes sense.

**Comment by ssanderson** [comment]
Pandas 0.17 will be harder I think.  We've had several reports of severe slowdowns (as in, order of magnitude slowdowns) from users on the google group who had upgraded pandas.

**Comment by richafrank** [comment]
Fixed by #969.

**Comment by twiecki** [comment]
:100:
